### DISCUSSION TITLE
Optimum/Correct Value for Neighbor Loader function in GraphSAGE

### DISCUSSION THREAD

**Discussion by Chaitanyasrini** [pyg-team/pytorch_geometric#8923] [General]
Hi, I am currently working on a project using GraphSAGE algorithm for node classification task. My dataset size is:
Number of nodes: 310 million
Number of edges: 660 million
I am using NeighborLoader function given by pytorch geometric for graphSAGE. In this I want to seek guidance on how much number of neighbors to sample with the dataset I am using which is mentioned above. What could be the optimal value for number of neighbors to sample from. In the research paper they are using [25,10] with K=2 hops. In some other online examples, they are using [10,5]. But I would like to know what would be the values I can use for my data of this size for my project.

**Comment by rusty1s** [comment]
The more neighbors and the more layers you sample, the slower your GNN will get, but usually sampling more neighbors leads to better model performance. How many neighbors to sample depends on your dataset. I would simply suggest trying different configurations.
